src/python/pants/core/goals/test.py

```diff
@@ -419,6 +424,11 @@ async def run_tests(
             status = "failed"
             exit_code = cast(int, result.exit_code)
         console.print_stderr(f"{sigil} {result.address} {status}.")
+        if result.extra_output and result.extra_output.files:
+            workspace.write_digest(
+                result.extra_output.digest,
+                path_prefix=os.path.join("dist", result.address.path_safe_spec),
```

Comment:
You've got `dist/<address>/<files>` here, but test coverage uses `dist/coverage/python/<files>`. It seems like `dist/<goal>/[not sure]` may be a good way to go. Too nested seems bad for project in one language. Not nested enough seems like it could cause problems for large repos or many languages repos. Since its out of scope for this PR and dist will need to be gone over for all cases with unavoidable fallout, perhaps its best to leave as you have it and batch the fallout into a dedicated dist cleanup project.

Reply:
D'oh re using DistDir , good catch.

I'll go for `dist/test/<address>/<files>` - adding the `test` goal disambiguator is good, and the `<address>` should be enough to avoid collisions (we merge coverage, so don't need the `<address>` there).

Reply:
But this does remind me that it's unclear how useful this feature is, since you end up with a separate html file for each test file.  There is no existing functionality to merge them (and pytest-html maintainers have resisted creating such a feature: https://github.com/pytest-dev/pytest-html/issues/48), but I suppose it wouldn't be too hard to do if someone wants to.